pytest/pytest.go

```diff
@@ -132,58 +134,155 @@ func RunBenchmarks(b *testing.B, testDir string) {
 
 // RunScript runs the provided path to a script.
 // RunScript captures the stdout and stderr while executing the script
-// and compares it to a golden file:
+// and compares it to a golden file, blocking until completion.
 //
 //	RunScript("./testdata/foo.py")
 //
 // will compare the output with "./testdata/foo_golden.txt".
 func RunScript(t *testing.T, fname string) {
+
+	RunTestTasks(t, []*TestTask{
+		{
+			PyFile: fname,
+		},
+	})
+}
+
+// RunTestTasks runs each given task in a newly created py.Context concurrently.
+// If a fatal error is encountered, the given testing.T is signaled.
+func RunTestTasks(t *testing.T, tasks []*TestTask) {
+	onCompleted := make(chan *TestTask)
+
+	numTasks := len(tasks)
+	for ti := 0; ti < numTasks; ti++ {
+		task := tasks[ti]
+		go func() {
+			err := task.run()
+			task.Error = err
+			onCompleted <- task
+		}()
+	}
+
+	tasks = tasks[:0]
+	for ti := 0; ti < numTasks; ti++ {
+		task := <-onCompleted
+		if task.Error != nil {
+			t.Error(task.Error)
+		}
+		tasks = append(tasks, task)
+	}
+}
+
+var (
+	taskCounter int32
+
+	// RegenGoldFiles will cause RunTestTasks() and RunScript() to overwrite their output as the "golden" file (rather than compare against it)
+	RegenGoldFiles bool
```

Comment:
what about making this a `flag.Bool(..., false, ...)` ?
like we do in:
https://github.com/gonum/plot/blob/master/cmpimg/checkplot.go#L19

as you can see, in there we are a bit clever: when provided with `"output.png"` as an output, we infer the golden file to be `"output_golden.png"`.
we could get smart like that.